Sources/Authgear.swift

```diff
@@ -615,17 +743,22 @@ public class Authgear: NSObject {
             uiLocales: uiLocales,
             wechatRedirectURI: wechatRedirectURI,
             page: page,
-            prefersEphemeralWebBrowserSession: prefersEphemeralWebBrowserSession
+            prefersEphemeralWebBrowserSession: prefersEphemeralWebBrowserSession,
+            useWebView: useWebView
         ), handler: handler)
     }
 
     private func authorize(
         _ options: AuthorizeOptions,
         handler: @escaping AuthorizeCompletionHandler
     ) {
-        let handler = self.withMainQueueHandler(handler)
-        self.workerQueue.async {
-            self.authorizeWithSession(options, handler: handler)
+        if options.useWebView == true {
+            self.authorizeWithoutSession(options, handler: handler)
+        } else {
+            let handler = self.withMainQueueHandler(handler)
+            self.workerQueue.async {
+                self.authorizeWithSession(options, handler: handler)
```

Comment:
The suffix WithSession and WithoutSession is not too self-explanatory. I suggest we rename them to `authorizeWithWKWebView` and `authorizeWithASWebAuthenticationSession`. Apply to `reauthenticateXXX` as well.